tests/py/dynamo/conversion/test_where_aten.py

```diff
@@ -76,6 +77,86 @@ def forward(self, condition):
             (condition,),
         )
 
+    # shape, min shape range, opt shape range, max shape range
+    @parameterized.expand(
+        [
+            (
+                "3d_condition_3d_xshape_3d_yshape",
+                (-1, -1, -1),
+                (1, 1, 1),
+                (1, 2, 3),
+                (3, 3, 3),
+                (-1, -1, -1),
+                (1, 1, 1),
+                (1, 2, 3),
+                (3, 3, 3),
+                (-1, -1, -1),
+                (1, 1, 1),
+                (1, 2, 3),
+                (3, 3, 3),
+            ),
+            (
+                "1d_condition_3d_xshape_2d_yshape",
+                (-1),
+                (1,),
+                (2,),
+                (4,),
+                (-1, -1, -1),
+                (1, 1, 1),
+                (3, 2, 2),
+                (3, 2, 4),
+                (
+                    -1,
+                    -1,
+                ),
+                (1, 1),
+                (2, 2),
+                (2, 4),
+            ),
+            (
+                "2d_condition_3d_xshape_2d_yshape",
+                (-1, -1),
+                (4, 1),
+                (4, 2),
+                (5, 4),
+                (-1, -1, -1),
+                (1, 1, 1),
+                (3, 1, 2),
+                (3, 1, 4),
+                (
+                    -1,
+                    -1,
+                ),
+                (4, 1),
+                (4, 2),
+                (5, 4),
+            ),
+        ]
+    )
+    def test_with_dynamic_shape(self, *args):
+        class Where(nn.Module):
+            def forward(self, condition, x, y):
+                return torch.ops.aten.where.self(condition, x, y)
+
+        input_specs = [
+            Input(
+                shape=args[1],
+                dtype=torch.bool,
+                shape_ranges=[(args[2], args[3], args[4])],
```

Comment:
Instead of `shape_ranges`, can you use `torch_tensorrt.Input(min_shape=<>, opt_shape=<> ...)` directly ?